src/kubernetes.cc

```diff
@@ -762,21 +762,23 @@ struct Watcher : public std::thread {
       }
       service_->stop();
     } else {
-      silence_timer_->RunAsyncAfter(std::chrono::seconds(
-          max_silence_seconds_),
-          [this](boost::system::error_code const &ec) {
-        if (ec != boost::asio::error::operation_aborted) {
-          service_->stop();
-        }
-      });
+      if (max_silence_seconds_ > 0 && silence_timer_ != nullptr) {
```

Comment:
you can just do `max_silence_seconds_ > 0 && silence_timer_` because of https://en.cppreference.com/w/cpp/memory/unique_ptr/operator_bool

Reply:
We've been explicit in other files such as json.cc and environment.cc, so I'll leave it as-is for now, thanks for pointing this out though!

Reply:
no problem - I think functionality-wise they are the same without much styling/consistency concern for me, so that's why I approved in advance.